vkcs/dc/resource_ip_port_forwarding.go

```diff
@@ -0,0 +1,376 @@
+			},
+
+			"created_at": schema.StringAttribute{
+				Computed:    true,
+				Description: "Creation timestamp",
+			},
+
+			"updated_at": schema.StringAttribute{
+				Computed:    true,
+				Description: "Update timestamp",
+			},
+
+			"region": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "The `region` to fetch availability zones from, defaults to the provider's `region`.",
+			},
+		},
+		Description: "Manages a direct connect ip port forwarding resource._note_ This resource requires Sprut SDN to be enabled in your project.",
+	}
+}
+
+func (r *IPPortForwardingResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.config = req.ProviderData.(clients.Config)
+}
+
+func (r *IPPortForwardingResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data IPPortForwardingResourceModel
+	diags := req.Plan.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	region := data.Region.ValueString()
+	if region == "" {
+		region = r.config.GetRegion()
+	}
+
+	networkingClient, err := r.config.NetworkingV2Client(region, networking.SprutSDN)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating VKCS networking client", err.Error())
+		return
+	}
+
+	ipPortForwardingCreateOpts := ipportforwardings.CreateOpts{
+		Name:          data.Name.ValueString(),
+		Description:   data.Description.ValueString(),
+		DCInterfaceID: data.DCInterfaceID.ValueString(),
+		Protocol:      data.Protocol.ValueString(),
+		Source:        data.Source.ValueString(),
+		Destination:   data.Destination.ValueString(),
+		Port:          int(data.Port.ValueInt64()),
+		ToDestination: data.ToDestination.ValueString(),
+		ToPort:        int(data.Port.ValueInt64()),
+	}
+
+	ipPortForwardingResp, err := ipportforwardings.Create(networkingClient, &ipportforwardings.IPPortForwardingCreate{IPPortForwarding: &ipPortForwardingCreateOpts}).Extract()
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vkcs_dc_ip_port_forwarding", err.Error())
+		return
+	}
+	ipPortForwardingID := ipPortForwardingResp.ID
+	resp.State.SetAttribute(ctx, path.Root("id"), ipPortForwardingID)
+
+	data.ID = types.StringValue(ipPortForwardingID)
+	data.Region = types.StringValue(region)
+	data.DCInterfaceID = types.StringValue(ipPortForwardingResp.DCInterfaceID)
+	data.Protocol = types.StringValue(ipPortForwardingResp.Protocol)
+	data.ToDestination = types.StringValue(ipPortForwardingResp.ToDestination)
+	data.Name = types.StringValue(ipPortForwardingResp.Name)
+	data.Description = types.StringValue(ipPortForwardingResp.Description)
+	data.CreatedAt = types.StringValue(ipPortForwardingResp.CreatedAt)
+	data.UpdatedAt = types.StringValue(ipPortForwardingResp.UpdatedAt)
+
+	if ipPortForwardingResp.Source != "" {
```

Comment:
Doesn't API return these attributes?

Reply:
Yes, if they are absent, API returns null

Reply:
Why shouldn't we update them then? Doesn't it bring some inconsistency when someone resets them to zero-value out of Terraform scope and we don't produce any plan changes? Of course, the point is not in the`Create` method, but in the `Read`.

Reply:
Rewrote to support null values